sw/host/ot_certs/src/x509.rs

```diff
@@ -0,0 +1,621 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{bail, Context, Result};
+use num_bigint_dig::BigUint;
+use std::marker::PhantomData;
+
+use openssl::asn1::{Asn1Integer, Asn1Object, Asn1OctetString, Asn1Time};
+use openssl::bn::{BigNum, BigNumContext};
+use openssl::ec::{EcGroup, EcKey};
+use openssl::ecdsa::EcdsaSig;
+use openssl::hash::MessageDigest;
+use openssl::nid::Nid;
+use openssl::pkey::PKey;
+use openssl::pkey::{Private, Public};
+use openssl::x509::extension;
+use openssl::x509::{X509Extension, X509NameBuilder, X509};
+
+use crate::offsets::{CertificateWithVariables, OffsetGenerator};
+use crate::template;
+
+struct Algorithm {
+    key_pair: PKey<Private>,
+    hash: MessageDigest,
+}
+
+fn ecgroup_from_curve(curve: &template::EcCurve) -> Result<EcGroup> {
+    match curve {
+        template::EcCurve::Prime256v1 => Ok(EcGroup::from_curve_name(Nid::X9_62_PRIME256V1)?),
+    }
+}
```

Comment:
Can `from_curve_name` fail if we're giving it that constant from the `openssl` crate?

If not, suggestion:

```rust
impl From<template::EcCurve> for EcGroup {
    fn from(ec_curve: template::EcCurve) -> EcGroup {
        match ec_curve {
            template::EcCurve::Prime256v1 => {
                EcGroup::from_curve_name(Nid::X9_62_PRIME256V1).unwrap()
            },
        }
    }
}
```

Reply:
It can fail from some constants that do not represent curves but I think for this one it cannot fail.